addons/mail/tests/discuss/test_message_controller.py

```diff
@@ -419,6 +419,30 @@ def test_mail_partner_from_email_authenticated(self):
             self.env["res.partner"].search_count([('email', '=', "john2@test.be")]),
             "'mail/message/post' does not create another user if there's already a user with matching email",
         )
+        # Handle partner creation when given name_addr emails as additional_values keys
+        res7 = self.url_open(
+            url="/mail/message/post",
+            data=json.dumps(
+                {
+                    "params": {
+                        "thread_model": "discuss.channel",
+                        "thread_id": self.channel.id,
+                        "post_data": {
+                            "body": "test",
+                        },
+                        "partner_emails": ["john3@test.be"],
+                        "partner_additional_values": {"john3 <john3@test.be>": {'phone': '123456789'}},
+                    },
+                }
+            ),
+            headers={"Content-Type": "application/json"},
+        )
+        self.assertEqual(res7.status_code, 200)
+        self.assertEqual(
+            1,
+            self.env["res.partner"].search_count([('email', '=', "john3@test.be"), ('phone', '=', "123456789")]),
+            "additional_values not handled correctly when using keys in name_addr format",
```

Comment:
:nerd_face: this way it's obvious what the intended behavior actually is
```suggestion
            "additional_values should be handled correctly when using keys in name_addr format",
```

Reply:
Can't you modify "res5" above, so that we directly test the "formatted" use case ? Knowing that if the complex one works, the simple one works ^^ . Less diff :)